code/atom.dm

```diff
@@ -615,6 +615,8 @@
 //mbc : sorry, i added a 'is_special' arg to this proc to avoid race conditions.
 /atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
 	if (user && W && !(W.flags & SUPPRESSATTACK))  //!( istype(W, /obj/item/grab)  || istype(W, /obj/item/spraybottle) || istype(W, /obj/item/card/emag)))
+		if(SEND_SIGNAL(src,COMSIG_ATTACKBY,W,user))
+			return
```

Comment:
I think this should be outside the if. SUPPERSSATTACK is (probably) meant to suppress just the message.

Reply:
Yeah, I agree. Did a shallow test, and things still work as expected too.